daml-lf/encoder/src/test/scala/com/digitalasset/daml/lf/archive/testing/DamlLfEncoderTest.scala

```diff
@@ -71,7 +73,18 @@ class DamlLfEncoderTest
     for {
       pkgWithId <- dar.main +: dar.dependencies
       (_, pkg) = pkgWithId
-      segments <- pkg.getDamlLf1.getModulesList.asScala.map(_.getName.getSegmentsList.asScala)
+      version = pkg.getMinor
+      internedStrings = pkg.getDamlLf1.getInternedStringsList.asScala.toArray
+      dottedNames = pkg.getDamlLf1.getInternedDottedNamesList.asScala.map(
+        _.getSegmentsInternedStrList.asScala.map(internedStrings(_))
+      )
+      segments <- pkg.getDamlLf1.getModulesList.asScala.map(
+        mod =>
+          if (version < "7")
+            mod.getNameDname.getSegmentsList.asScala
```

Comment:
This is going to be break once we reach 1.10, i.e., "10" < "7".

Reply:
Thanks. I will fix that.

Reply:
fixed

Reply:
Great!